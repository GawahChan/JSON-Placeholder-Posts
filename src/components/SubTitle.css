.subTitle {
    display: flex;
    flex-direction: row;
    background-color: #6868c5;
    border: 1px solid #DDDDDD;
    padding: 10px;
}

.subTitle h2 {
    flex: 1;
    margin: 0;
    text-align: center;
    font-weight: 500;
    color: #DDDDDD
}

.btn-displayPosts {
    width: 0; 
    height: 0; 
    border-left: 10px solid transparent;
    border-right: 10px solid transparent;
    cursor: pointer;
}

.btn-displayPosts:hover {
    animation: pulse 3s infinite; 
}

.btn-displayPosts.up{
    border-top: none;
    border-bottom: 10px solid #DDDDDD;
}

.btn-displayPosts.down {
    border-top: 10px solid #dddddd;
    border-bottom: none;
}

@keyframes pulse {
    0% {
        transform: scale(1);
    }
    50% {
        transform: scale(1.6);
    }
    100% {
        transform: scale(1);
    }
}