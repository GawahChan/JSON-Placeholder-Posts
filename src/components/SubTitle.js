import React from 'react'
import './SubTitle.css';

function SubTitle(props) {
    return (
        <div className='subTitle'>
            <h2>{props.title}</h2>
            <div className={`btn-displayPosts ${props.display ? 'up' : 'down'}`}
                onClick={props.toggleDisplay}
            />
        </div>
    );
}

export default SubTitle;