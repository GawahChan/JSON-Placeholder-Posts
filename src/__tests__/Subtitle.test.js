import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import SubTitle from '../components/SubTitle';

describe('SubTitle', () => {
    const subTitleTree = renderer.create(<SubTitle />).toJSON();
    const subTitle = shallow(<SubTitle />);

    it('renders correctly', () => {
        expect(subTitleTree).toMatchSnapshot();
    });

    it('checks if <div className=subTitle /> exists', () => {
        expect(subTitle.find('.subTitle').exists()).toBe(true);
    });
});