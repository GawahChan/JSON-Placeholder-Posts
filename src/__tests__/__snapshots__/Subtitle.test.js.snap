// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubTitle renders correctly 1`] = `
<div
  className="subTitle"
>
  <h2 />
  <div
    className="btn-showPosts down"
  />
</div>
`;
